tests/preprocessing/test_data_preprocessor.py

```
@@ -0,0 +1,190 @@
+    with pytest.raises(FileNotFoundError):
+        DataProcessor(config=project_config)
+
+    mock_logger.warning.assert_called_once()
+    mock_logger.info.assert_called_once()
+    mock_logger.error.assert_called_once_with(mocker.ANY)
+
+def test_load_data_fallback_csv_read_error(project_config, mock_uci_fetch_fail, mock_logger, mock_read_csv, mock_path, mocker):
+    mock_path.return_value.exists.return_value = True
+    mock_read_csv.side_effect = Exception("Error reading CSV")
+
+    with pytest.raises(Exception):
```

Ruff

`pytest.raises(Exception)` should be considered evil

(B017)